### ISSUE TITLE
51% Starting Java Language Server... never starts

### ISSUE THREAD

**Issue by ScrippsFEE** [redhat-developer/vscode-java#433]
Opening this new issue because #162 is closed. 
I can't get this plugin working. I've cleaned the workspace as instructed in issue #162 

##### Environment
- Operating System: OSX Sierra
- JDK version: "1.8.0_65"
- Visual Studio Code version: 1.19.3
- Java extension version: 0.18.1

##### Steps To Reproduce
1. start VS Code
2. wait for JLS to start

[attach a sample project reproducing the error]
[attach logs](https://github.com/redhat-developer/vscode-java/wiki/Troubleshooting#enable-logging)

##### JPS and JSTACK Output
```
Identity added: /Users/162297/bitbucket/git-key (/Users/162297/bitbucket/git-key)
$ jps
47571 org.eclipse.equinox.launcher_1.5.0.v20180119-0753.jar
29205 GradleDaemon
52997 org.eclipse.equinox.launcher_1.5.0.v20180119-0753.jar
55814 Jps
53575 org.eclipse.equinox.launcher_1.5.0.v20180119-0753.jar
55772 org.eclipse.equinox.launcher_1.5.0.v20180119-0753.jar
55101 org.eclipse.equinox.launcher_1.5.0.v20180119-0753.jar

$ jstack 55101
2018-02-06 13:19:46
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.65-b01 mixed mode):

"Framework stop" #30 prio=5 os_prio=31 tid=0x00007ff9eba05800 nid=0x1007 waiting on condition [0x0000700005eb2000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x0000000787b040e0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at java.util.concurrent.ThreadPoolExecutor.awaitTermination(ThreadPoolExecutor.java:1465)
        at org.gradle.internal.concurrent.ManagedExecutorImpl.stop(ManagedExecutorImpl.java:81)
        at org.gradle.internal.concurrent.DefaultExecutorFactory$TrackedManagedExecutor.stop(DefaultExecutorFactory.java:77)
        at org.gradle.internal.concurrent.ManagedExecutorImpl.stop(ManagedExecutorImpl.java:72)
        at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:98)
        at org.gradle.internal.concurrent.DefaultExecutorFactory.stop(DefaultExecutorFactory.java:32)
        at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:98)
        at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectProvider.stop(DefaultServiceRegistry.java:592)
        at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:98)
        at org.gradle.internal.service.DefaultServiceRegistry$OwnServices.stop(DefaultServiceRegistry.java:518)
        at org.gradle.internal.service.DefaultServiceRegistry$CachingProvider.stop(DefaultServiceRegistry.java:1017)
        at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:98)
        at org.gradle.internal.service.DefaultServiceRegistry.close(DefaultServiceRegistry.java:265)
        - locked <0x0000000783a361e0> (a java.lang.Object)
        at org.gradle.tooling.internal.consumer.ConnectorServices.reset(ConnectorServices.java:54)
        at com.gradleware.tooling.toolingclient.internal.DefaultToolingClient.expireDaemons(DefaultToolingClient.java:292)
        at com.gradleware.tooling.toolingclient.internal.DefaultToolingClient.stop(DefaultToolingClient.java:244)
        at org.eclipse.buildship.core.CorePlugin.stop(CorePlugin.java:122)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$4.run(BundleContextImpl.java:835)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$4.run(BundleContextImpl.java:1)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.stop(BundleContextImpl.java:828)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.stopWorker0(EquinoxBundle.java:1016)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.stopWorker(EquinoxBundle.java:359)
        at org.eclipse.osgi.container.Module.doStop(Module.java:636)
        at org.eclipse.osgi.container.Module.stop(Module.java:498)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.decStartLevel(ModuleContainer.java:1723)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1642)
        - locked <0x0000000780353628> (a java.lang.Object)
        at org.eclipse.osgi.container.SystemModule.stopWorker(SystemModule.java:270)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule.stopWorker(EquinoxBundle.java:192)
        at org.eclipse.osgi.container.Module.doStop(Module.java:636)
        at org.eclipse.osgi.container.Module.stop(Module.java:498)
        at org.eclipse.osgi.container.SystemModule.stop(SystemModule.java:202)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule$1.run(EquinoxBundle.java:210)
        at java.lang.Thread.run(Thread.java:745)

"Attach Listener" #29 daemon prio=9 os_prio=31 tid=0x00007ff9ecb97000 nid=0x5e07 waiting on condition[0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Connection worker" #28 prio=5 os_prio=31 tid=0x00007ff9ebb2d800 nid=0xc707 runnable [0x0000700005cad000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
        at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
        at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:103)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
        - locked <0x0000000785b07f58> (a sun.nio.ch.Util$2)
        - locked <0x0000000785b07f48> (a java.util.Collections$UnmodifiableSet)
        - locked <0x0000000785b07e28> (a sun.nio.ch.KQueueSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
        at org.gradle.messaging.remote.internal.inet.SocketConnection$SocketInputStream.read(SocketConnection.java:154)
        at com.esotericsoftware.kryo.io.Input.fill(Input.java:139)
        at com.esotericsoftware.kryo.io.Input.require(Input.java:159)
        at com.esotericsoftware.kryo.io.Input.readInt(Input.java:308)
        at org.gradle.internal.serialize.kryo.KryoBackedDecoder.readSmallInt(KryoBackedDecoder.java:120)
        at org.gradle.internal.serialize.DefaultSerializerRegistry$TaggedTypeSerializer.read(DefaultSerializerRegistry.java:80)
        at org.gradle.internal.serialize.Serializers$StatefulSerializerAdapter$1.read(Serializers.java:36)
        at org.gradle.messaging.remote.internal.inet.SocketConnection.receive(SocketConnection.java:77)
        at org.gradle.launcher.daemon.client.DaemonClientConnection.receive(DaemonClientConnection.java:79)
        at org.gradle.launcher.daemon.client.DaemonClient.executeBuild(DaemonClient.java:145)
        at org.gradle.launcher.daemon.client.DaemonClient.execute(DaemonClient.java:125)
        at org.gradle.launcher.daemon.client.DaemonClient.execute(DaemonClient.java:80)
        at org.gradle.tooling.internal.provider.DaemonBuildActionExecuter.execute(DaemonBuildActionExecuter.java:52)
        at org.gradle.tooling.internal.provider.DaemonBuildActionExecuter.execute(DaemonBuildActionExecuter.java:35)
        at org.gradle.tooling.internal.provider.LoggingBridgingBuildActionExecuter.execute(LoggingBridgingBuildActionExecuter.java:63)
        at org.gradle.tooling.internal.provider.LoggingBridgingBuildActionExecuter.execute(LoggingBridgingBuildActionExecuter.java:35)
        at org.gradle.tooling.internal.provider.ProviderConnection.run(ProviderConnection.java:132)
        at org.gradle.tooling.internal.provider.ProviderConnection.run(ProviderConnection.java:108)
        at org.gradle.tooling.internal.provider.DefaultConnection.getModel(DefaultConnection.java:168)
        at org.gradle.tooling.internal.consumer.connection.CancellableModelBuilderBackedModelProducer.produceModel(CancellableModelBuilderBackedModelProducer.java:53)
        at org.gradle.tooling.internal.consumer.connection.PluginClasspathInjectionSupportedCheckModelProducer.produceModel(PluginClasspathInjectionSupportedCheckModelProducer.java:41)
        at org.gradle.tooling.internal.consumer.connection.AbstractConsumerConnection.run(AbstractConsumerConnection.java:58)
        at org.gradle.tooling.internal.consumer.connection.ParameterValidatingConsumerConnection.run(ParameterValidatingConsumerConnection.java:46)
        at org.gradle.tooling.internal.consumer.DefaultModelBuilder$1.run(DefaultModelBuilder.java:60)
        at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:84)
        at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
        at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:58)
        at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
        at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:55)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.lang.Thread.run(Thread.java:745)

"Worker-3" #26 prio=5 os_prio=31 tid=0x00007ff9ea8ce800 nid=0xc503 in Object.wait() [0x0000700005bab000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:197)
        - locked <0x000000078108e4b0> (a org.eclipse.core.internal.jobs.WorkerPool)
        at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:239)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

"Worker-2: Compacting resource model" #25 prio=5 os_prio=31 tid=0x00007ff9ea97c000 nid=0xc303 in Object.wait() [0x0000700005aa8000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.core.internal.jobs.ThreadJob.waitForRun(ThreadJob.java:313)
        - locked <0x0000000781406838> (a java.lang.Object)
        at org.eclipse.core.internal.jobs.ThreadJob.joinRun(ThreadJob.java:202)
        at org.eclipse.core.internal.jobs.ImplicitJobs.begin(ImplicitJobs.java:92)
        at org.eclipse.core.internal.jobs.JobManager.beginRule(JobManager.java:308)
        at org.eclipse.core.internal.utils.StringPoolJob.run(StringPoolJob.java:98)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:60)

"Bundle File Closer" #24 daemon prio=5 os_prio=31 tid=0x00007ff9e7ab0800 nid=0xc103 in Object.wait() [0x00007000059a5000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x000000078400c9d8> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"pool-1-thread-1" #23 prio=5 os_prio=31 tid=0x00007ff9e6b06800 nid=0xbf03 waiting on condition [0x00007000058a1000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
        at java.lang.Thread.sleep(Native Method)
        at org.eclipse.jdt.ls.core.internal.JobHelpers.waitForJobs(JobHelpers.java:159)
        at org.eclipse.jdt.ls.core.internal.JobHelpers.waitForInitializeJobs(JobHelpers.java:142)
        at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.didChangeConfiguration(JDTLanguageServer.java:242)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:51)
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint$$Lambda$32/1297836716.apply(Unknown Source)
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:126)
        at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:172)
        at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:143)
        at org.eclipse.jdt.ls.core.internal.ParentProcessWatcher.lambda$0(ParentProcessWatcher.java:83)
        at org.eclipse.jdt.ls.core.internal.ParentProcessWatcher$$Lambda$29/620557167.consume(UnknownSource)
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:149)
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:77)
        at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:84)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

"pool-2-thread-1" #22 prio=5 os_prio=31 tid=0x00007ff9e7b06800 nid=0xbd03 waiting on condition [0x000070000579f000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x0000000781009190> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1081)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

"Worker-1: Updating Maven Dependencies" #20 prio=5 os_prio=31 tid=0x00007ff9e7ae4000 nid=0xbb03 in Object.wait() [0x000070000569c000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.core.internal.jobs.ThreadJob.waitForRun(ThreadJob.java:313)
        - locked <0x0000000781406838> (a java.lang.Object)
        at org.eclipse.core.internal.jobs.ThreadJob.joinRun(ThreadJob.java:202)
        at org.eclipse.core.internal.jobs.ImplicitJobs.begin(ImplicitJobs.java:92)
        at org.eclipse.core.internal.jobs.JobManager.beginRule(JobManager.java:308)
        at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob$1.call(ProjectRegistryRefreshJob.java:96)
        at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob$1.call(ProjectRegistryRefreshJob.java:1)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
        at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:1355)
        at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob.run(ProjectRegistryRefreshJob.java:81)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:60)

"Java indexing" #19 daemon prio=4 os_prio=31 tid=0x00007ff9e6aa4800 nid=0xb903 in Object.wait() [0x0000700005599000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x0000000781188798> (a org.eclipse.jdt.internal.core.search.indexing.IndexManager)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.jdt.internal.core.search.processing.JobManager.run(JobManager.java:375)
        - locked <0x0000000781188798> (a org.eclipse.jdt.internal.core.search.indexing.IndexManager)
        at java.lang.Thread.run(Thread.java:745)

"Worker-0: Initialize Workspace" #18 prio=5 os_prio=31 tid=0x00007ff9e993e800 nid=0xb707 waiting on condition [0x0000700005494000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x0000000787b05438> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
        at java.util.concurrent.ArrayBlockingQueue.take(ArrayBlockingQueue.java:403)
        at org.gradle.tooling.internal.consumer.BlockingResultHandler.getResult(BlockingResultHandler.java:40)
        at org.gradle.tooling.internal.consumer.DefaultModelBuilder.get(DefaultModelBuilder.java:50)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.eclipse.buildship.core.workspace.internal.ConnectionAwareLauncherProxy.invokeRun(ConnectionAwareLauncherProxy.java:170)
        at org.eclipse.buildship.core.workspace.internal.ConnectionAwareLauncherProxy.invoke(ConnectionAwareLauncherProxy.java:160)
        at com.sun.proxy.$Proxy39.get(Unknown Source)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider$2.get(DefaultModelProvider.java:122)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider$3.call(DefaultModelProvider.java:142)
        at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4724)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3522)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2315)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2278)
        - locked <0x0000000787b05378> (a com.google.common.cache.LocalCache$StrongEntry)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2193)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3932)
        at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4721)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider.getFromCache(DefaultModelProvider.java:153)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider.executeOperation(DefaultModelProvider.java:138)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider.executeModelBuilder(DefaultModelProvider.java:118)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider.fetchModels(DefaultModelProvider.java:81)
        at org.eclipse.buildship.core.workspace.internal.DefaultModelProvider.fetchEclipseGradleProjects(DefaultModelProvider.java:99)
        at org.eclipse.jdt.ls.core.internal.managers.GradleProjectImporter.startSynchronization(GradleProjectImporter.java:150)
        at org.eclipse.jdt.ls.core.internal.managers.GradleProjectImporter.importDir(GradleProjectImporter.java:97)
        at org.eclipse.jdt.ls.core.internal.managers.GradleProjectImporter.lambda$0(GradleProjectImporter.java:89)
        at org.eclipse.jdt.ls.core.internal.managers.GradleProjectImporter$$Lambda$44/1390632210.accept(Unknown Source)
        at java.util.ArrayList.forEach(ArrayList.java:1249)
        at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1080)
        at org.eclipse.jdt.ls.core.internal.managers.GradleProjectImporter.importToWorkspace(GradleProjectImporter.java:89)
        at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.importProjects(ProjectsManager.java:105)
        at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.access$1(ProjectsManager.java:99)
        at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager$1.run(ProjectsManager.java:93)
        at org.eclipse.jdt.internal.core.BatchOperation.executeOperation(BatchOperation.java:38)
        at org.eclipse.jdt.internal.core.JavaModelOperation.run(JavaModelOperation.java:732)
        at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2214)
        at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2241)
        at org.eclipse.jdt.core.JavaCore.run(JavaCore.java:5676)
        at org.eclipse.jdt.core.JavaCore.run(JavaCore.java:5633)
        at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.initializeProjects(ProjectsManager.java:86)
        at org.eclipse.jdt.ls.core.internal.handlers.InitHandler$1.runInWorkspace(InitHandler.java:193)
        at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:39)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:60)

"Worker-JM" #17 prio=5 os_prio=31 tid=0x00007ff9e6ab7800 nid=0xb613 in Object.wait() [0x0000700005393000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.InternalWorker.run(InternalWorker.java:59)
        - locked <0x000000078108e658> (a java.util.ArrayList)

"Start Level: Equinox Container: 55aec47a-c11a-443a-b3d9-4dd40aaa7bfb" #16 daemon prio=5 os_prio=31 tid=0x00007ff9e9907000 nid=0x7903 in Object.wait() [0x0000700005290000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x000000078108e670> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x000000078108e670> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"Framework Event Dispatcher: Equinox Container: 55aec47a-c11a-443a-b3d9-4dd40aaa7bfb" #15 daemon prio=5 os_prio=31 tid=0x00007ff9e6981800 nid=0x7703 in Object.wait() [0x000070000518d000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x00000007811885f0> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"Active Thread: Equinox Container: 55aec47a-c11a-443a-b3d9-4dd40aaa7bfb" #13 prio=5 os_prio=31 tid=0x00007ff9ea88d800 nid=0x7503 waiting on condition [0x000070000508a000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000007803496e0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

"process reaper" #12 daemon prio=10 os_prio=31 tid=0x00007ff9e690a800 nid=0x7303 waiting on condition[0x0000700004f87000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x0000000780390810> (a java.util.concurrent.SynchronousQueue$TransferStack)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460)
        at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
        at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1066)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

"Service Thread" #10 daemon prio=9 os_prio=31 tid=0x00007ff9e685d000 nid=0x6f03 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread3" #9 daemon prio=9 os_prio=31 tid=0x00007ff9e9045800 nid=0x6d03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread2" #8 daemon prio=9 os_prio=31 tid=0x00007ff9e902c800 nid=0x6b03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread1" #7 daemon prio=9 os_prio=31 tid=0x00007ff9e902c000 nid=0x6903 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #6 daemon prio=9 os_prio=31 tid=0x00007ff9e7843000 nid=0x6703 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #5 daemon prio=9 os_prio=31 tid=0x00007ff9e7842800 nid=0x6503 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Surrogate Locker Thread (Concurrent GC)" #4 daemon prio=9 os_prio=31 tid=0x00007ff9e7841800 nid=0x6203 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=31 tid=0x00007ff9e9020000 nid=0x5203 in Object.wait() [0x00007000046c1000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
        - locked <0x000000078034a3d0> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
        at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

"Reference Handler" #2 daemon prio=10 os_prio=31 tid=0x00007ff9e901f000 nid=0x5003 in Object.wait() [0x00007000045be000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:157)
        - locked <0x0000000780390a68> (a java.lang.ref.Reference$Lock)

"main" #1 prio=5 os_prio=31 tid=0x00007ff9e9802000 nid=0x1b03 in Object.wait() [0x0000700002e78000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.osgi.container.SystemModule.waitForStop(SystemModule.java:168)
        - locked <0x000000078055fed8> (a java.util.concurrent.atomic.AtomicReference)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle.waitForStop(EquinoxBundle.java:295)
        at org.eclipse.osgi.launch.Equinox.waitForStop(Equinox.java:181)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.shutdown(EclipseStarter.java:444)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:262)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
        at org.eclipse.equinox.launcher.Main.main(Main.java:1471)

"VM Thread" os_prio=31 tid=0x00007ff9e7830000 nid=0x4e03 runnable

"Gang worker#0 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e900a800 nid=0x2403 runnable

"Gang worker#1 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e980b000 nid=0x2603 runnable

"Gang worker#2 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e980b800 nid=0x2803 runnable

"Gang worker#3 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e900b000 nid=0x2a03 runnable

"Gang worker#4 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e900c000 nid=0x2c03 runnable

"Gang worker#5 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e980c000 nid=0x2e03 runnable

"Gang worker#6 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e980c800 nid=0x3003 runnable

"Gang worker#7 (Parallel GC Threads)" os_prio=31 tid=0x00007ff9e900c800 nid=0x3203 runnable

"G1 Main Concurrent Mark GC Thread" os_prio=31 tid=0x00007ff9e981d800 nid=0x4603 runnable

"Gang worker#0 (G1 Parallel Marking Threads)" os_prio=31 tid=0x00007ff9e981e000 nid=0x4803 runnable

"Gang worker#1 (G1 Parallel Marking Threads)" os_prio=31 tid=0x00007ff9e981f000 nid=0x4a03 runnable

"G1 Concurrent Refinement Thread#0" os_prio=31 tid=0x00007ff9e980e000 nid=0x4403 runnable

"G1 Concurrent Refinement Thread#1" os_prio=31 tid=0x00007ff9e980d800 nid=0x4203 runnable

"G1 Concurrent Refinement Thread#2" os_prio=31 tid=0x00007ff9e9011000 nid=0x4003 runnable

"G1 Concurrent Refinement Thread#3" os_prio=31 tid=0x00007ff9e9010000 nid=0x3e03 runnable

"G1 Concurrent Refinement Thread#4" os_prio=31 tid=0x00007ff9e900f800 nid=0x3c03 runnable

"G1 Concurrent Refinement Thread#5" os_prio=31 tid=0x00007ff9e900e800 nid=0x3a03 runnable

"G1 Concurrent Refinement Thread#6" os_prio=31 tid=0x00007ff9ea000000 nid=0x3803 runnable

"G1 Concurrent Refinement Thread#7" os_prio=31 tid=0x00007ff9e900e000 nid=0x3603 runnable

"G1 Concurrent Refinement Thread#8" os_prio=31 tid=0x00007ff9e900d000 nid=0x3403 runnable

"String Deduplication Thread" os_prio=31 tid=0x00007ff9e682e800 nid=0x4c03 runnable

"VM Periodic Task Thread" os_prio=31 tid=0x00007ff9e6855800 nid=0x7103 waiting on condition

JNI global references: 344
```

**Comment by silkster** [comment]
I have uninstalled every plugin related to Java. I reinstalled the Language Support for Java plugin.  This error persists. Anyone here able to help?

**Comment by silkster** [comment]
Now it says:

> 51% Starting Java Language Server - Refreshing '/jdt.ls-java-project'.

**Comment by fbricon** [comment]
@silkster do you have a new thread dump with the server being stuck at ` Refreshing '/jdt.ls-java-project'.`

Does it happen for all your projects or just 1 in particular?

**Comment by fbricon** [comment]
Can you try to kill all your java processes before starting vscode (i.e gradle and org.eclipse.equinox.launcher_1.5.0.v20180119-0753.jar). Make sure all vscode instances are closed or else the processes will be respawned

**Comment by ScrippsFEE** [comment]
I wanted to follow up with this issue. I ended up getting a new MacBook and found that I don't have this issue anymore so it was definitely something environment specific on my old MacBook.

**Comment by ScrippsFEE** [comment]
@fbricon I only work on 1 Java project, but the problem definitely persisted. We use Docker which was also very problematic on my old Mac compared to the new one so it could have been something to do with Java or JDK specifically. Thanks for your time and help!
